### PR TITLE
Add transactions file and update animals table.

### PR DESCRIPTION
**Pull request by ShahierNashaat**
Pull request for review vet database using Mysql

- Add a column species of type string to your animals' table.

- Insert the following animals
    - Insert Animal: His name is Charmander. He was born on Feb 8th, 2020,  and currently weighs -11kg. He is not neutered and he has never tried to escape.
    - Insert Animal: Her name is Plantmon. She was born on Nov 15th, 2022, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
    - Insert Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to 3 times.
    - Insert Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
    - Insert Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
    - Insert Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.

- Create transactions file.
    - update the animals' table by setting the species column to unspecified. Then roll back the change.
    - Update the animals' table by setting the species column to digimon for all animals that have a name ending in mon. Update the animals' table by setting the species column to pokemon for all animals that don't have species already set.
    - delete all records in the animals' table, then roll back the transaction.
    - Delete all animals born after Jan 1st, 2022. Create a savepoint for the transaction. Update all animals' weight to be their weight multiplied by -1. Rollback to the savepoint. Update all animals' weights that are negative to be their weight multiplied by -1.

- Execute the following queries.
    - Find the number of animals.
    - Find the number of animals that have never tried to escape.
    - Find the average weight of animals.
    - Find the most escaped_attempts if it was neutered or not in animals.
    - Find the min and max weight of every species in animals.
    - Find the average escape attempts for every species that was born between 1990 and 2000.

### CODE REVIEW

File: transactions.sql [hunk 1]
```diff
@@ -0,0 +1,39 @@
+-- update the animals table by setting the species column to unspecified.
+-- Then roll back the change.
+
+BEGIN;
+update animals set species = 'unspecified';
+ROLLBACK;
```
Comment (SajjadAhmad14) [inline transactions.sql]: Please add a query to verify explicitly that change was made before and after rollback.

File: transactions.sql [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+
+-- Update the animals table by setting the species column to digimon for all animals 
+-- that have a name ending in mon.
+-- Update the animals table by setting the species column to pokemon for all animals 
+-- that don't have species already set.
+
+BEGIN;
+update animals set species = 'digimon' where name like '%mon';
+update animals set species = 'pokemon' where species is NULL;
```
Comment (SajjadAhmad14) [inline transactions.sql:19]: Please add a query to verify explicitly that change was made before and after the commit.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Shahier, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git flow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by descholar-ceo — APPROVED** [review]
Hi @ShahierNashaat and @SajjadAhmad14 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**